### DISCUSSION TITLE
How does a hyper client open multiple H2 connections for the same address?

### DISCUSSION THREAD

**Discussion by silence-coding** [hyperium/hyper#3174] [Q&A] [answered]
How does a hyper client open multiple H2 connections for the same address? Check whether only one TCP connection becomes a bottleneck in the case of high concurrency.

**Comment by seanmonstar** [comment, marked as answer]
hyper currently does **not** open more than one connection within the same client, based on what [RFC 9113 9.1](https://httpwg.org/specs/rfc9113.html#rfc.section.9.1) says:

> Clients SHOULD NOT open more than one HTTP/2 connection to a given host and port pair.
